[SPARK-38351][TESTS] Don't use deprecate symbol API in test classes

martin-g opened the pull request:
### What changes were proposed in this pull request?

Replace symbols like `'abc` with the more verbose `Symbol("abc") in the test code. 

### Why are the changes needed?

Building with Scala 2.13 produces a lot of warnings like the following ones:

```
[warn] /home/runner/work/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/execution/BaseScriptTransformationSuite.scala:562:11: [deprecation @  | origin= | version=2.13.0] symbol literal is deprecated; use Symbol("d") instead
[warn]           'd.cast("string"),
[warn]           ^
[warn] /home/runner/work/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/execution/BaseScriptTransformationSuite.scala:563:11: [deprecation @  | origin= | version=2.13.0] symbol literal is deprecated; use Symbol("e") instead
[warn]           'e.cast("string")).collect())
```

This should make it easier to upgrade to Scala 3 later.


### Does this PR introduce _any_ user-facing change?

No! The PR touches only test classes!

### How was this patch tested?

The build at CI must be green!

martin-g committed:
[SPARK-38242][SS] Sort the SparkSubmit debug output

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
[SPARK-38351][TESTS] Scala 2.13 deprecated 'name symbols

The recommended way is to use Symbol("name").
Fix deprecation warnings only in the test suites

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

srowen reviewed (commented):
Yeah I think we did this for non-test code earlier. It should be OK as it is functionally identical and avoids a 2.13 deprecation warning.

martin-g commented on core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala:
```diff
@@ -906,10 +906,10 @@ private[spark] class SparkSubmit extends Logging {
     if (args.verbose) {
       logInfo(s"Main class:\n$childMainClass")
-      logInfo(s"Arguments:\n${childArgs.mkString("\n")}")
+      logInfo(s"Arguments:\n${childArgs.sorted.mkString("\n")}")
```
This is because I've created the branch for this PR from the branch of #35556. I will remove this commit from here.

martin-g commented:
@dongjoon-hyun @HyukjinKwon What is your opinion ? Would you merge such PR ?

How big should the PR be ? Fix all issues of this type in the whole or module by module ?

srowen commented:
Heh, no right answer, but I'd say fix < 200 files at a time? maybe that gets to all test code? that would be a fine scope

AmplabJenkins commented:
Can one of the admins verify this patch?

HyukjinKwon commented:
There was a previous try to refer to (https://github.com/apache/spark/pull/31569). I agree that we should probably make it go away anyway .. cc @sarutak FYI

HyukjinKwon commented:
There was a previous try to refer to (https://github.com/apache/spark/pull/31569) cc @sarutak FYI

sarutak commented:
For making it easy to upgrade to Scala 3, another option would be to import `language.deprecated.symbolLiterals._` when we support Scala 3.
What do you think, all?

srowen commented:
Does that just suppress the warning? I think just consistently using Symbol() is easier IMHO; there may not be that much left to change.

sarutak commented:
Yeah, I thought it's better to replace them but there were some objections to do it in the previous PR. So, I just suggest another possible solution.

martin-g commented:
I wasn't aware of #31569. There some people said that it might be better to migrate to `s"abc"` instead of `Symbol("abc")`, and I agree.
I am fine with any of the following:
* close this PR as "Won't do"
* finish it
* re-work it to `s""`

srowen commented:
Does `s"..."` do anything here? I thought that just expressed interpolated strings

martin-g commented:
Sorry! Muscle memory ... I meant `$"abc"` - https://github.com/wangyum/spark/commit/b428acb1322f534ad6f028c23286cf2ce18277f6

martin-g commented:
I will create a JIRA ticket and update the PR soon!

martin-g committed:
SPARK-38351: Migrate few more entries from 'abc to Symbol("abc")

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
The build fails due to :
```
 java.lang.RuntimeException: Failing because of negative scalastyle result
[error] 	at scala.sys.package$.error(package.scala:30)
[error] 	at org.scalastyle.sbt.Tasks$.handleResult$1(Plugin.scala:132)
[error] 	at org.scalastyle.sbt.Tasks$.doScalastyleWithConfig$1(Plugin.scala:187)
...
```

```
[error] /__w/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala:580: File line length exceeds 100 characters
[error] /__w/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala:927: File line length exceeds 100 characters
...
```

But executing `./dev/scalafmt` leads to many other unrelated formatting changes like:
```diff
import org.apache.spark.sql.catalyst.expressions.{InSet, Literal, NamedExpression}
-import org.apache.spark.sql.catalyst.util.DateTimeTestUtils.{outstandingTimezonesIds, outstandingZoneIds}
+import org.apache.spark.sql.catalyst.util.DateTimeTestUtils.{
+  outstandingTimezonesIds,
+  outstandingZoneIds
+}
 import org.apache.spark.sql.catalyst.util.DateTimeUtils
 import org.apache.spark.sql.execution.ProjectExec
 import org.apache.spark.sql.functions._
@@ -43,83 +46,54 @@ class ColumnExpressionSuite extends QueryTest with SharedSparkSession {
   import testImplicits._
 
   private lazy val booleanData = {
-    spark.createDataFrame(sparkContext.parallelize(
-      Row(false, false) ::
-      Row(false, true) ::
-      Row(true, false) ::
-      Row(true, true) :: Nil),
+    spark.createDataFrame(
+      sparkContext.parallelize(
+        Row(false, false) ::
+          Row(false, true) ::
+          Row(true, false) ::
+          Row(true, true) :: Nil),
```

It might take me a while to format manually just the related code.

martin-g committed:
SPARK-38351: Replace some more 'abc with Symbol("abc")

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

srowen commented:
Yeah don't bother with scalafmt here, just fix the long lines if there are only a few instances

martin-g committed:
SPARK-38351: Manually format the code to not exceed 100 chars per line

E.g.:
[error] /__w/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala:927: File line length exceeds 100 characters
[error] /__w/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala:928: File line length exceeds 100 characters

./dev/scalafmt formats other unrelated lines too, so it is not really
usable

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@srowen The PR is ready for review!

srowen commented:
Merged to master

HeartSaVioR commented:
Just to see our preference, given `$"..."` is not mentioned at all for deprecation and it is simpler than `Symbol("...")`, why not consistently using `$"..."`? My intuition is that `'` would be used as it is simplest representation to refer a column, and `$"..."` is going to be simplest once we are enforced to use `Symbol("...")` instead.

srowen commented:
Is that the same thing? $".." is shorthand for a Column, not Scala Symbol

HeartSaVioR commented:
Right, but what else we use Symbol except representing Column? I might be missing something, but from what I've seen from test codes, majority of usage has been the same for $"...".

srowen commented:
Hm, I think you have a point. We definitely need Symbol in some parts of the Scala code, but I think they're rare and limited to parts that reason about closures, etc. For columns, yeah, in hindsight I agree.

Eh, hm, martin this might be my fault here. It may be smarter to turn all of those changes into $"..." after all if they're all really column refs. I totally missed that. We could do it in a follow up or revert this or whatever's easy, if you're willing.

HeartSaVioR commented:
Follow-up fix sounds good to me. Easier to work based on this since we now just need to try finding `Symbol(` and replace there. Finding shorthand of Symbol `'` would be much more bothering work despite warning message will guide us.

martin-g commented:
Agreed! I will work on this soon!

jiangxb1987 commented:
Do we still have any tests for `'`? Since this is still supported, we should have at least one test to make sure it's working and if it's broken by a future Scala version, we can notice it.

martin-g commented:
@jiangxb1987 Yes, there are still usages of `'` in the tests. This PR changed just part of the test files to keep the diff not too big,
I didn't have time for the follow-up in the last two weeks but I will do it soon!

srowen commented:
@martin-g any chance you have some time to take another look at this?

martin-g commented:
I promise to do it real soon! I have something else to finish first! 
The new PR will be just for `master` branch, i.e. `3.4.x`.

martin-g commented:
@srowen https://github.com/apache/spark/pull/35976
